libraries/AP_RangeFinder/RangeFinder.cpp

```diff
@@ -494,11 +494,8 @@ void RangeFinder::detect_instance(uint8_t instance)
             hal.i2c_mgr->get_device(0, AP_RANGEFINDER_PULSEDLIGHTLRF_ADDR)));
     } 
     if (type == RangeFinder_TYPE_MBI2C) {
-        if (AP_RangeFinder_MaxsonarI2CXL::detect(*this, instance)) {
-            state[instance].instance = instance;
-            drivers[instance] = new AP_RangeFinder_MaxsonarI2CXL(*this, instance, state[instance]);
-            return;
-        }
+        add_backend(AP_RangeFinder_MaxsonarI2CXL::detect(*this, instance, state[instance],
+            hal.i2c_mgr->get_device(0, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR)));
```

Comment:
Is there any reason on why passing `hal.i2c_mgr->get_device(0, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR)` as an argument and not initializing with this value by default in `AP_RangeFinder_MaxsonarI2CXL` constructor?